src/web/routes/entities/index.ts

```diff
@@ -23,6 +29,28 @@ type AssociateEntityResponse = Response<
 	{ connectInstallation: ConnectInstallation }
 >;
 
+export type DisassociateEntityRequestParams = Omit<
+	DisassociateEntityUseCaseParams,
+	'atlassianUserId' | 'connectInstallation'
+>;
+
+type DisassociateEntityResponse = Response<
+	{ design: AtlassianDesign } | string,
+	{ connectInstallation: ConnectInstallation }
+>;
+
+const getUserIdHeaderOrThrow = (
+	userId: string | string[] | undefined,
+	res: Response,
+): string => {
+	if (!userId || typeof userId !== 'string') {
+		const errorMessage = 'Missing or invalid User-Id header';
+		res.status(401).send(errorMessage);
+		throw errorMessage;
```

Comment:
**issue:** Currently, `getUserIdHeaderOrThrow` does 2 things:

1. Sends an error response to a client
2. Throws an exception

I think we need to do either #1 or #2 but not both since it can cause tricky issues with request processing (e.g., when an error handler attempts to write to an already sent response). Also, it is better to avoid throwing a non-`Error` (e.g., `string`) to avoid type-related issues in error handling code. 

To address this, please, consider throwing a specific error (e.g., `BadRequestError`), which we could handle in `errorHandlerMiddleware`:

```ts
// web/errors.ts
export class BadRequestError extends Error {}


// src/web/routes/entities/index.ts
const getUserIdHeaderOrThrow = (
	userId: string | string[] | undefined,
	res: Response,
): string => {
	if (!userId || typeof userId !== 'string') {
		throw new BadRequestError('Missing or invalid User-Id header');
	}
	return userId;
}

// web/middleware/error-handler-middleware.ts
export const errorHandlerMiddleware = (
	err: Error,
	req: Request,
	res: Response,
	next: NextFunction,
): void => {
	// ...

	if (err instanceof BadRequesError) {
		res.status(HttpStatusCode.BadRequest).send(err.message);
	} 

	// ...
};
```

Reply:
Yeah great suggestion, this has simplified the code nicely as well.